relayer/src/supervisor.rs

```diff
@@ -119,11 +127,76 @@ impl Supervisor {
         })
     }
 
+    fn create_workers(&mut self) -> Result<(), BoxError> {
+        let req = QueryChannelsRequest {
+            pagination: ibc_proto::cosmos::base::query::pagination::all(),
+        };
+
+        // For each opened channel spawn:
+        // - the path worker for AtoB (used to clear packets) and
+        // - the client worker for the A side client of the channel's connection (used for refresh)
+        // TODO - we should move to one supervisor, operating on a set of chains:
+        // `for chain_a in self.chains.clone().iter() {`, lookup chain b in registry based
+        // on the client chain_id, etc.
+        for chains in [self.chains.clone(), self.chains.clone().swap()].iter() {
+            let channels: Vec<IdentifiedChannelEnd> = chains.a.query_channels(req.clone())?;
+            for channel in channels.iter() {
+                // only clear packets and start client workers for opened channels
+                if !channel
+                    .channel_end
+                    .state_matches(&ibc::ics04_channel::channel::State::Open)
```

Comment:
We can perhaps introduce a `is_open` method on `ChannelEnd`, and use `filter(ChannelEnd::is_open)` on the iterator?